nbtlib/tag.py

```diff
@@ -118,8 +118,11 @@ def write_numeric(fmt, value, buff, byteorder='big'):
 def read_string(buff, byteorder='big'):
     """Read a string from a file-like object."""
     length = read_numeric(USHORT, buff, byteorder)
-    return buff.read(length).decode('utf-8')
-
+    br=buff.read(length)
+    try:
+        return br.decode('utf-8')
+    except UnicodeDecodeError:
+        return br
 
 def write_string(value, buff, byteorder='big'):

```

Comment:
forget to modify `write_string` too. :cry:
if a key of a Compound cannot be decoded,It may lead to error without modify `write_string`.